org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenImpl.java

```diff
@@ -484,15 +484,10 @@ public Map<File, MavenExecutionResult> readMavenProjects(Collection<File> pomFil
     return result;
   }
 
-  /**
-   * Makes MavenProject instances returned by #readProject methods suitable for caching and reuse with other
-   * MavenSession instances.<br/>
-   * Do note that MavenProject.getParentProject() cannot be used for detached MavenProject instances. Use
-   * #resolveParentProject to resolve parent project instance.
-   */
+  @Deprecated
```

Comment:
The deprecation annotation should have the for removal flag set and java-doc should be added that this method is not necessary anymore and one can safely cache all projects (but should somebody do that?).

Reply:
It is already in the interface, Eclipse just complained that if its deprecated in the interface it should be deprecated in the impl...